controllers/snow_machineconfig_controller.go

```diff
@@ -0,0 +1,110 @@
+package controllers
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+	"sigs.k8s.io/cluster-api/util/patch"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/aws"
+	"github.com/aws/eks-anywhere/pkg/providers/snow"
+)
+
+// SnowMachineConfigReconciler reconciles a SnowMachineConfig object
+type SnowMachineConfigReconciler struct {
+	Client        client.Client
+	Log           logr.Logger
+	ClientBuilder aws.ClientBuilder
+	Validator     *snow.Validator
+}
+
+func NewSnowMachineConfigReconciler(client client.Client, log logr.Logger, clientBuilder aws.ClientBuilder, validator *snow.Validator) *SnowMachineConfigReconciler {
```

Comment:
I think we should try to receive interfaces defined in this file if possible

Reply:
Got it, moved ClientBuilder interface to this file and the Validator will be dynamically created in reconcile